src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/RequireEmptyLineBeforeAtClauseBlockCheck.java

```diff
@@ -0,0 +1,197 @@
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks that one blank line before the at-clauses block if it is present in Javadoc.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations
+ * if the Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">
+ * Tight-HTML Rules</a>.
+ * Type is {@code boolean}. Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;RequireEmptyLineBeforeAtClauseBlock&quot;/&gt;
+ * </pre>
+ * <p>
+ * By default, the check will report a violation if there is no blank line before the at-clause,
+ * like in the example below.
+ * </p>
+ * <pre>
+ * &#47;**
+ *  * testMethod's javadoc.
+ *  * &#64;return something (violation)
+ *  *&#47;
+ * public boolean testMethod() {
+ *     return false;
+ * }
+ * </pre>
+ * <p>
+ *  Valid javadoc should have a blank line separating the parameters, return, throw, or
+ *  other at-clauses like in the example below.
+ *  </p>
+ *  <pre>
+ *  &#47;**
+ *  * testMethod's javadoc.
+ *  *
+ *  * &#64;param firstParam
+ *  * &#64;return something
+ *  *&#47;
+ *  public boolean testMethod(int firstParam) {
+ *      return false;
+ *  }
+ *  </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.line.before}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.35
+ */
+@StatelessCheck
+public class RequireEmptyLineBeforeAtClauseBlockCheck extends AbstractJavadocCheck {
+
+    /**
+     * The key in "messages.properties" for the message that describes an at-clause in javadoc
+     * requiring an empty line before it.
+     */
+    public static final String MSG_JAVADOC_TAG_LINE_BEFORE = "javadoc.tag.line.before";
+
+    /**
+     * When javadoc has only a javadoc tag like {@literal @} in it, the JAVADOC_TAG in a JAVADOC
+     * detail node will always have 3 siblings before it. The parse tree looks like:
+     * <pre>
+     * JAVADOC
+     * \
+     *  -- NEWLINE
+     *  -- LEADING_ASTERISK
+     *  -- WS
+     *  -- JAVADOC_TAG
+     * </pre>
+     */
+    private static final int NUM_OF_SIBLINGS_BEFORE_ONLY_JAVADOC_TAG = 3;
```

Comment:
After a bit of think on this number ... I still think it is bit non-reliable approach.
please 
1) move whole javadoc of this field to `isOnlyAtClauseNothingElse`.
2) please format tree to follow our style as at https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html#JAVADOC
3) please avoid NUM_OF_SIBLINGS_BEFORE_ONLY_JAVADOC_TAG as constant and let use logic like:
```
prevSibling = getPrevSibling
if (prevSibling != null && prevSibling.getType=WS) {
  prevSibling = getPrevSibling
  if (prevSibling != null && prevSibling.getType=LEADING_ASTERISK) {
```
(or something similar)

Reply:
I tried to get rid of the constant but I failed and actual did worse! I added a second constant.
My first attempt was the code below:
```java
boolean result = false;
DetailNode prevClause = JavadocUtil.getPreviousSibling(tagNode);
if (prevClause != null && prevClause.getType() == JavadocTokenTypes.WS) {
    prevClause = JavadocUtil.getPreviousSibling(tagNode);
    if (prevClause != null && prevClause.getType() == JavadocTokenTypes.LEADING_ASTERISK) {
        prevClause = JavadocUtil.getPreviousSibling(tagNode);
        result = prevClause.getType() == JavadocTokenTypes.NEWLINE;
    }
}
return result;
```
but that result in many uncovered branches. While searching for test cases to cover the missing branches, I found this test case that I needed to cover
```java
/**
 *@param nospace There's no space.
 */
```
I need to make sure the above example does not recommend a newline. Another indepedent check will recommend adding a space.
Supporting this case requires even more branches.
Using hardcode lists of the expected types came out much cleaner.
Let me know what you think. I will try again with if statements and search for test cases to cover all the branches is creates if you don't like the List solution.

Reply:
it is very good that coverage tool try to help us with extra test cases and we become to see more edge case.
We should not afraid coverage ... it guides to better code and tests.
I still scary of this magic number `3`.
let have one more PR and we will compare code to be more certain about alternative to magic number.

> I need to make sure the above example does not recommend a newline. Another indepedent check will recommend adding a space.

yes. yes.

Reply:
The magic number 3 is gone. It's replaced by the two expected lists of previous siblings:
```java
    private static final List<Integer> ONLY_TAG_VARIATION_1 = Arrays.asList(
            JavadocTokenTypes.WS,
            JavadocTokenTypes.LEADING_ASTERISK,
            JavadocTokenTypes.NEWLINE);

    /**
     * Case when no space separates the tag and the asterisk like in the below example.
     * <pre>
     *  /**
     *   *&#64;param noSpace there is no space here
     * </pre>
     */
    private static final List<Integer> ONLY_TAG_VARIATION_2 = Arrays.asList(
            JavadocTokenTypes.LEADING_ASTERISK,
            JavadocTokenTypes.NEWLINE);
```